packages/fx-core/src/component/generator/generatorAction.ts

```diff
@@ -56,119 +50,21 @@ export interface GeneratorAction {
-  name: GeneratorActionName.FetchTemplateZipFromSourceCode,
-  run: (context: GeneratorContext) => {
-    const isDebugMode = () => {
-      const DebugTemplateFlag = process.env[FeatureFlagName.DebugTemplate];
-      return DebugTemplateFlag?.toLowerCase() === "true" && process.env.NODE_ENV === "development";
-    };
-
-    if (!isDebugMode()) {
-      return Promise.resolve();
-    }
-
-    if (context.zip) {
-      return Promise.resolve();
-    }
-
-    context.logProvider.debug(`Fetching template zip from source code: ${JSON.stringify(context)}`);
-    //! This path only works in debug mode
-    const templateSourceCodePath = path.resolve(
-      __dirname,
-      "../../../../../",
-      "templates",
-      context.language!
-    );
-
-    context.zip = zipFolder(templateSourceCodePath);
-    return Promise.resolve();
-  },
-};
-
-export const fetchSampleInfoAction: GeneratorAction = {
-  name: GeneratorActionName.FetchSampleInfo,
-  run: async (context: GeneratorContext) => {
-    const sample = await getSampleInfoFromName(context.name);
-    context.sampleInfo = sample.downloadUrlInfo;
-  },
-};
-
-export const downloadDirectoryAction: GeneratorAction = {
-  name: GeneratorActionName.DownloadDirectory,
-  run: async (context: GeneratorContext) => {
-    context.logProvider.debug(`Downloading sample by directory: ${JSON.stringify(context)}`);
-    if (!context.sampleInfo) {
-      throw new MissKeyError("sampleInfo");
-    }
-
-    context.outputs = await downloadDirectory(context.sampleInfo, context.destination);
-    if (!context.outputs?.length) {
-      throw new SampleNotFoundError(context.name);
-    }
-  },
-};
-
-export const fetchUrlForHotfixOnlyAction: GeneratorAction = {
-  name: GeneratorActionName.FetchUrlForHotfixOnly,
-  run: async (context: GeneratorContext) => {
-    if (context.zip || context.cancelDownloading) {
-      return;
-    }
-
-    context.logProvider.debug(`Fetching template url with tag: ${JSON.stringify(context)}`);
-    const latestTag = await getTemplateLatestTag(
-      context.language!,
-      context.tryLimits,
-      context.timeoutInMs
-    );
-    const localVer = getTemplateLocalVersion();
-    const latestVer = latestTag.split("@")[1];
-    // git tag version is higher than the local version, download template from github
-    if (semver.gt(latestVer, localVer)) {
-      context.url = getTemplateZipUrlByTag(context.language!, latestTag);
-    } else {
-      // generate template from fallback
-      context.cancelDownloading = true;
-    }
-  },
-};
-
-export const fetchZipFromUrlAction: GeneratorAction = {
-  name: GeneratorActionName.FetchZipFromUrl,
+export const RemoteTemplateAction: GeneratorAction = {
+  name: GeneratorActionName.RemoteTemplate,
   run: async (context: GeneratorContext) => {
-    if (context.zip || context.cancelDownloading) {
-      return;
-    }
+    const templateUrl = await determineTemplateSource(context);
+    if (templateUrl === "local") return;
```

Comment:
For coding style, it is better to add {} even it is a simple return;

Reply:
Ok, I'll fix it.